### PR TITLE
245 hfp optimization

### PR DESCRIPTION
**Pull request by danieldunderfelt**
Mostly schema updates related to https://github.com/HSLdevcom/bultti/issues/245

Backend PR: https://github.com/HSLdevcom/bultti/pull/247

### CODE REVIEW

**Review by Divinare — CHANGES_REQUESTED** [review]
Just two comments, otherwise LGTM!

File: src/inspection/InspectionConfig.tsx [hunk 1]
```diff
@@ -49,6 +50,15 @@ const InspectionConfig: React.FC<PropTypes> = observer(({ saveValues, inspection
     })
   }, [])
 
+  let onChangeInspectionDate = useCallback((startDate: Date, endDate: Date, id?: string) => {
```
[the remark this answers is not in the document]
Comment (danieldunderfelt) [reply src/inspection/InspectionConfig.tsx:53]: The ID is undefined for pre-inspections.

File: src/inspection/inspectionDate/InsectionDateList.tsx [hunk 2]
```diff
@@ -119,24 +119,21 @@ const InspectionDateList: React.FC<PropTypes> = observer(
...
-            )}
-            {inspectionDates.map((inspectionDate: InspectionDate) => (
-              <InspectionDateListItem
-                key={inspectionDate.id}
-                removeItem={removeItem}
-                inspectionDate={inspectionDate}
-              />
-            ))}
-          </>
+        <LoadingDisplay loading={isLoading} />
```
[the remark this answers is not in the document]
Comment (danieldunderfelt) [reply src/inspection/inspectionDate/InsectionDateList.tsx:122]: No, the whole list was jumping around and flashing when refreshing, it was very annoying. I'll add a check for "isLoading" to the empty message so it won't show when loading.

File: src/inspection/InspectionConfig.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Divinare) [reply src/inspection/InspectionConfig.tsx:53]: Ah yeah... Forgot that InspectionSelectDates serves both pre and post inspections

**Review by Divinare — APPROVED** [review]

**Pull request merged**